zeebe/bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/CompensationEventDefinitionValidator.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.model.bpmn.validation.zeebe;
+
+import io.camunda.zeebe.model.bpmn.impl.BpmnModelConstants;
+import io.camunda.zeebe.model.bpmn.instance.Activity;
+import io.camunda.zeebe.model.bpmn.instance.CompensateEventDefinition;
+import io.camunda.zeebe.model.bpmn.instance.SubProcess;
+import io.camunda.zeebe.model.bpmn.util.ModelUtil;
+import org.camunda.bpm.model.xml.validation.ModelElementValidator;
+import org.camunda.bpm.model.xml.validation.ValidationResultCollector;
+
+public class CompensationEventDefinitionValidator
+    implements ModelElementValidator<CompensateEventDefinition> {
+
+  @Override
+  public Class<CompensateEventDefinition> getElementType() {
+    return CompensateEventDefinition.class;
+  }
+
+  @Override
+  public void validate(
+      final CompensateEventDefinition compensateEventDefinition,
+      final ValidationResultCollector validationResultCollector) {
+
+    final String activityRef =
+        compensateEventDefinition.getAttributeValue(BpmnModelConstants.BPMN_ATTRIBUTE_ACTIVITY_REF);
+    final Activity referencedActivity = compensateEventDefinition.getActivity();
+
+    if (referencedActivity == null) {
+      if (activityRef != null && !activityRef.isEmpty()) {
+        validationResultCollector.addError(
+            0,
+            String.format("The referenced compensation activity '%s' doesn't exist", activityRef));
+      }
+
+    } else {
+      if (!isValidCompensationActivity(referencedActivity)) {
```

Comment:
🔧 You can move the else branch into a separate method (eg.`validateReferencedCompensation` ) to improve the readability